src/input_handler.cpp

```diff
@@ -0,0 +1,52 @@
+#include "input_handler.h"
+
+namespace arch {
+    InputHandler InputHandler::input_handler_instance;
```

Comment:
Please avoid global initialization. Change it to `InputHandler InputHandler::input_handler_instance;` to `InputHandler InputHandler::input_handler_instance = nullptr` and look at comment for `InputHandler::get`